### PR TITLE
Create load_metadata method in IM

### PR DESCRIPTION
**Pull request by ew3361zh**
This PR seeks to address issue #607 to create a new function called `load_metadata()` in the IM class that loads the metadata from a JSON file.

The `example_metadata` file here is from Appendix 1 of the IM design doc.
Unit testing for IM is also moved to its own dedicated testing file.

### CODE REVIEW

File: RUFAS/input_manager.py [hunk 1]
```diff
@@ -17,3 +18,20 @@ def __new__(cls):
...
+        self.__metadata: Dict[str, Any] = {}
+
+    def _load_metadata(self, metadata_path: str = "input/example_metadata.json") -> None:
+        """
+        Loads metadata from json file to IM metadata object
+
+        Parameters
+        ----------
+            metadata_path : str
+                The path to the metadata file
```
Comment (Isgotkowitz) [inline RUFAS/input_manager.py]:
```suggestion
        metadata_path : str
            The path to the metadata file
```

**Review by Isgotkowitz — APPROVED** [review]
Looks good Niko. Only noticed one small formatting issue in docstring.

File: RUFAS/input_manager.py [hunk 2]
```diff
@@ -17,3 +18,26 @@ def __new__(cls):
     def __init__(self) -> None:
         if InputManager.__instance is None:
             InputManager.__instance = self
+        self.__metadata: Dict[str, Any] = {}
+
+    def _load_metadata(self, metadata_path: str = "input/example_metadata.json") -> None:
+        """
+        Loads metadata from json file to IM metadata object
```
Comment (PooyaHekmati) [inline RUFAS/input_manager.py]:
```suggestion
        Loads metadata from json file to IM metadata dict.
```

File: RUFAS/input_manager.py [hunk 3]
```diff
@@ -17,3 +18,26 @@ def __new__(cls):
...
+        Exception
+            If an error occurs while opening or reading the metadata_path file.
+
+        """
+        try:
+            with open(metadata_path) as metadata_file:
+                self.__metadata = json.load(metadata_file)
+
+        except Exception as e:
+            raise e
```
Comment (PooyaHekmati) [inline RUFAS/input_manager.py]: IM requirement 6:
> IM must report its activities’ logs to Output Manager’s logs pool.

Accessing IO is an important activity that deserves logging.

**Review by PooyaHekmati — APPROVED** [review]
Please address comments before merging.

**Pull request merged**